test/eventlistener_test.go

```diff
@@ -0,0 +1,114 @@
+// +build e2e
+
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"testing"
+
+	knativetest "github.com/knative/pkg/test"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestEventListenerCreate(t *testing.T) {
+	c, namespace := setup(t)
+	t.Parallel()
+
+	defer tearDown(t, c, namespace)
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+
+	t.Log("Start EventListener e2e test")
+
+	// Create EventListener
+	el, err := c.TriggersClient.TriggersV1alpha1().EventListeners(namespace).Create(
+		&v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-eventlistener",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				TriggerBindingRefs: []v1alpha1.TriggerBindingRef{
+					{
+						Name: "my-triggerbinding",
+					},
+				},
+			},
+		},
+	)
+	if err != nil {
+		t.Fatalf("Failed to create EventListener: %s", err)
+	}
+	t.Logf("Created EventListener %s in namespace %s", el.Name, el.Namespace)
+
+	// Verify the EventListener's Deployment is created
+	if err = wait.PollImmediate(interval, timeout, func() (bool, error) {
+		_, err := c.KubeClient.AppsV1().Deployments(namespace).Get(el.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return false, nil
+		}
+		return true, err
+	}); err != nil {
+		t.Fatalf("Failed to create EventListener Deployment: %s", err)
+	}
+	t.Logf("Found EventListener's Deployment")
+
+	// Verify the EventListener's Service is created
```

Comment:
something we haven't discussed (i dont think) - would we expect the EventListener's status section to have info about the resulting deployment + what endpoint it could be reached at? I'm thinking maybe yes? in which case instead of polling for everything, we could poll only for the status of the EventListener and then that should imply the existence of the rest of these resources - and it would be an error if they didnt exist - no polling required!

Reply:
Good point! I created a new issue [here](https://github.com/tektoncd/triggers/issues/54). Please update the issue if you have more to add on the subject 👍